pipeline/lib/pipeline.ml

```diff
@@ -151,12 +151,29 @@ let docker_make_bench ~run_args ~repository image =
         "opam exec -- make bench";
       ]
 
+let record_pipeline_stage ~stage ~serial_id ~conninfo image =
+  let+ job_id = Current_util.get_job_id image
+  and+ state = Current.state image in
+  match (job_id, state) with
```

Comment:
Isn't the intent to record metadata when the build starts or the job starts? It looks like nothing would be recorded unless there's an error in the job, given the patterns being matched here?

Reply:
So the way ocurrent works is that you can have three states an ongoing job, 
`  | Ok _ -> when the job has finished
  | Error (`Active _) -> when the job is active
  | Error (`Msg m) -> When there is an error running the job`. So we start recording when the job is active :)

Reply:
Ah, okay. Thanks for the explanation. The DB updates weren't happening because of the missing UNIQUE constraint, and I didn't correctly debug the problem. :)